x-pack/legacy/plugins/integrations_manager/index.ts

```diff
@@ -75,12 +75,13 @@ const pluginOptions: LegacyPluginOptions = {
   init(server: Server) {
     server.plugins.xpack_main.registerFeature(feature);
 
-    const coreSetup: CoreSetup = {
-      http: {
-        route: server.route.bind(server),
-      },
-    };
+    // convert hapi instance to KbnServer
+    // `kbnServer.server` is the same hapi instance
+    // `kbnServer.newPlatform` has important values
+    const kbnServer = (server as unknown) as KbnServer;
     const initializerContext: PluginInitializerContext = {};
+    const coreSetup: CoreSetup = kbnServer.newPlatform.setup.core;
```

Comment:
:+1: now you're using the actual new platform instead of approximating it with a shim.